frontend/src/metabase-lib/queries/structured/Filter.ts

```diff
@@ -61,6 +63,38 @@ export default class Filter extends MBQLClause {
     return this._query.removeFilter(this._index);
   }
 
+  betterDateLabel() {
+    const dimension = this.dimension();
+    const unit = dimension?.temporalUnit();
+    const op = this.operatorName();
+    const args = this.arguments();
+    const allStrs = args.every(arg => typeof arg === "string");
+    const isDate = ["day", "week", "month", "quarter", "year"].includes(unit);
+    const betweenDates = op === "between" && isDate;
+    const equalsWeek = op === "=" && unit === "week";
+    // modifying some of DEFAULT_DATE_FORMATS
+    const sliceFormats = {
+      "hour-of-day": "[hour] H",
+      "minute-of-hour": "[minute] m",
+      "day-of-month": "Do [day of month]",
+      "day-of-year": "DDDo [day of year]",
+      "week-of-year": "wo [week of year]",
+    };
+    if (allStrs) {
```

Comment:
Nit: we can removed a layer of nesting and make the code a bit more explicit by inverting this condition and returning early

```typescript
if (!allStrs) {
  return undefined;
}
```

It may even be better to include this right after `allStrs` is defined, that way the reader knows that all variables defined below that point are only relevant to the case where `allStrs` is `true`.

Reply:
dfde816929f2f1d74caa3d4c87f454bba23227e7